scipion/scripts/kickoff.py

```diff
@@ -421,33 +422,42 @@ def getTemplate(root):
         If more than one template is found or passed, a dialog is raised
         to choose one.
     """
-    templates = []
     templateFolder = getExternalJsonTemplates()
     customTemplates = len(sys.argv) > 1
     if customTemplates:
+        tl = TemplateList()
+        templates = []
         candidates = sys.argv[1:]
         for candFile in candidates:
             if os.path.isfile(candFile):
                 templates.append(String(candFile))
             else:
                 print(" > %s file does not exist." % candFile)
+        templates = tl.genFromStrList(templates).templates
     else:
-        # Check if there is any .json.template in the template folder
-        # get the template folder
-        for file in glob.glob1(templateFolder, "*.json.template"):
-            templates.append(String(file))
-
-    if len(templates):
-        if len(templates) == 1:
-            chosen = templates[0].get()
+        templates = []
+        # Check if other plugins have json.templates
+        templateFolder = pw.Config.getExternalJsonTemplates()
+        # Check if other plugins have json.templates
+        domain = pw.Config.getDomain()
+        tempList = TemplateList()
+        for pluginName, plugin in domain.getPlugins().items():
+            tDir = TemplateDirData(plugin)
+            if tDir.path:
+                Plugin.getTemplates(tempList, tDir, pluginName, templateFolder)
```

Comment:
we should take the plugin from plugin --> plugin.Plugin.getTemplates() and we do not need any of the parameters. All of them can be inferred within the Plugin context?